multinet/uploaders/csv.py

```diff
@@ -1,54 +1,27 @@
 """Multinet uploader for CSV files."""
 import csv
-import re
 from flasgger import swag_from
 from io import StringIO
-from dataclasses import dataclass
 
 from multinet import db, util
-from multinet.errors import AlreadyExists, FlaskTuple, ServerError, ValidationFailed
+from multinet.auth.util import require_writer
+from multinet.errors import AlreadyExists, FlaskTuple, ServerError
 from multinet.util import decode_data
-from multinet.validation import ValidationFailure, DuplicateKey, UnsupportedTable
+from multinet.validation.validate_csv import validate_csv
 
 from flask import Blueprint, request
 from flask import current_app as app
 from webargs import fields as webarg_fields
 from webargs.flaskparser import use_kwargs
 
 # Import types
-from typing import Set, MutableMapping, Sequence, Any, List, Dict
+from typing import Any, List, Dict
 
 
 bp = Blueprint("csv", __name__)
 bp.before_request(util.require_db)
 
 
-@dataclass
-class InvalidRow(ValidationFailure):
-    """Invalid syntax in a CSV file."""
-
-    row: int
-    fields: List[str]
-
-
-@dataclass
-class KeyFieldAlreadyExists(ValidationFailure):
-    """CSV file has both existing _key field and specified key field."""
-
-    key: str
-
-
-@dataclass
-class KeyFieldDoesNotExist(ValidationFailure):
-    """The specified key field does not exist."""
-
-    key: str
-
-
-class MissingBody(ValidationFailure):
-    """Missing body in a CSV file."""
-
-
 class CSVReadError(ServerError):
```

Comment:
Why was this error not moved with the rest of validation errors?

Reply:
Probably just a mistake, but I will double check and report back here.

Reply:
This was left in place because it's only used in the CSV uploader's `upload()` method, and is not really a validation error. It's a runtime error that can occur if there's an error reading the file in, which is a concern of the API method itself, not of the deeper validation methods.